server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/SqlPlanTask.java

```diff
@@ -163,6 +164,47 @@ protected boolean doStart(JobContext context) throws Exception {
         return true;
     }
 
+    private List<OffsetString> getSplitSql() {
+        SqlCommentProcessor processor = ConnectionSessionUtil.getSqlCommentProcessor(connectionSession);
+        StringBuffer buffer = new StringBuffer();
+        List<OffsetString> splitSqlList = new ArrayList<>();
+
+        if (CollectionUtils.isNullOrEmpty(parameters.getSqlObjectIds())) {
+            String sqlContent = parameters.getSqlContent();
+            log.info("SQL content={}", sqlContent);
+
+            List<OffsetString> offsetStringList = processor.split(buffer, sqlContent);
+            result.setTotalStatements(offsetStringList.size());
+
+            return offsetStringList;
+        }
+
+        CloudObjectStorageService cloudObjectStorageService = getCloudObjectStorageService();
+
+        if (Objects.isNull(cloudObjectStorageService) || !cloudObjectStorageService.supported()) {
+            log.warn("Cloud object storage service not supported.");
+            throw new UnexpectedException("Cloud object storage service not supported");
+        }
+
+        for (String objectId : parameters.getSqlObjectIds()) {
+            try {
+                byte[] bytes = cloudObjectStorageService.readContent(objectId);
+                if (Objects.isNull(bytes)) {
+                    log.warn("Read content from cloud storage returned null, objectId={}", objectId);
+                    continue;
+                }
+                log.info("successfully read content from cloud storage, objectId={}", objectId);
+                String sqlContent = new String(bytes, StandardCharsets.UTF_8);
+                List<OffsetString> offsetStringList = processor.split(buffer, sqlContent);
+                splitSqlList.addAll(offsetStringList);
```

Comment:
read all sql statement into memory may cost a lot of memory, 
the database change task already contains stream read/split logic, sql plan may follow the same flow.

Reply:
Already changed,  the sql plan follow the same logic as stream reading.